hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -1138,13 +1138,14 @@ private class Emit(
         val unified = impl.unify(args.map(_.typ))
         assert(unified)
 
+        val argPTypes = args.map(_.pType)
         val meth =
-          methods(fn).filter { case (argt, _) => argt.zip(args.map(_.typ)).forall { case (t1, t2) => t1 isOfType t2 } } match {
+          methods(fn).filter { case (argt, _) => argt.zip(argPTypes).forall { case (t1, t2) => t1 == t2 } } match {
```

Comment:
would it be safer to use isOfType here?

Reply:
no, that would actually be incorrect. 

`methods` is a cache of already-generated code. If the physical type of the arguments differs (though the virtual type is the same), we'll require a different generated implementation.

Reply:
Ah thank you. Since this is a more restrictive check, will it have side effects?